src/Sylius/Bundle/ApiBundle/Validator/Constraints/PromotionCouponEligibilityValidator.php

```diff
@@ -59,14 +59,23 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
-        /** @var PromotionCouponInterface $promotionCoupon */
+        /** @var PromotionCouponInterface|null $promotionCoupon */
         $promotionCoupon = $this->promotionCouponRepository->findOneBy(['code' => $value->couponCode]);
 
         /** @var OrderInterface $cart */
         $cart = $this->orderRepository->findCartByTokenValue($value->getOrderTokenValue());
 
         $cart->setPromotionCoupon($promotionCoupon);
 
+        if ($promotionCoupon === null) {
```

Comment:
Cannot we just add this check to the next `if` statement? They do the same thing

Reply:
Yes, but I decided add new message `can_not_be_blank `. I think it will be more clear

Reply:
Wellll, in fact it cannot be done this way 😄 The coupon code is not blank, it's provided by the customer - it's just invalid 💃 The blankness of the coupon is checked a few lines above